Source/Core/Common/VR/DolphinVR.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2016 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <cstring>
+
+#include "Common/VR/DolphinVR.h"
+#include "Common/VR/VRBase.h"
+#include "Common/VR/VRInput.h"
+#include "Common/VR/VRMath.h"
+#include "Common/VR/VRRenderer.h"
+
+namespace Common::VR
+{
+Base* s_module_base = NULL;
```

Comment:
These are still static variables.  This should be wrapped in a class and the functions should be member functions..

Reply:
I can change it to a singleton class. The problem here is that it is called from several project parts. Would you be fine with that?

Reply:
Singleton is still a static.  The point is to get rid of the global variable.  This really needs to be a class owned by Dolphin main.  In your current implementation it'd probably be owned by the OGL graphics library but I'm questioning if this is a great path forward.  Currently reading and thinking of how we should do this..

Reply:
I am afraid there is not a good solution. The only owner of the class could be Main. Then we would have to pass it as a parameter to many interfaces/implementations. I am not sure if that would go under 100 modified files.

Reply:
We're currently migrating over to getting rid of all global variables. If you take a look at at Core/System.h, that's a class where we're kind of cramming in references to all class instances that would otherwise just be globals or singletons. Would this class fit in there? If so, you can piggyback on all of our existing efforts to pass the System instance around as a parameter everywhere.

Reply:
I thought of that.  I think there's some bigger issues here though.  Like how this is currently tangled with our `OpenGL` renderer implementation which makes me wonder what it means if we want to add an additional renderer backend in the future (AFAIK OpenXR is not tied to a specific render implementation).

Reply:
OpenXR is tight to specific renderer backends but it supports almost all of the mainstream backends, from what I know only Apple Metal is missing.

The specific OpenXR code is needed on:
* renderer initialization
* binding backbuffer
* start frame rendering
* end frame rendering

The main problem with OpenXR concept is that implements multiple core technologies so you can hardly put it on specific component place. For most calls you need an instance of XrSession or XrInstance. So which have to be shared between rendering/input/app flow.